python/startup/Python/Startup/__init__.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+
+def bootstrap_sgtk():
+	"""
+	Bootstraps SGTK to Nuke Studio, or Hiero, depending on which
+	mode it's currently running in.
+	"""
+	import nuke
+
+	if nuke.env.get("hiero"):
+		_bootstrap_hiero()
+	elif nuke.env.get("studio"):
+		_bootstrap_nuke_studio()
+
+def _clean_env():
+	"""
+	Cleans up SGTK related environment variables.
+	"""
+	for var in ["TANK_ENGINE", "TANK_CONTEXT", "TANK_FILE_TO_OPEN"]:
+		if var in os.environ:
+			del os.environ[var]
+
+def _setup_sgtk(output_handle):
+	"""
+	Extracts the necessary information from the environment and starts
+	the tk-nuke engine.
+	"""
+	try:
+		import tank
+	except Exception, e:
+		output_handle("Shotgun: Could not import sgtk! Disabling: %s" % e)
```

Comment:
With a generic exception catch like this, it is safer to end with ```% str(e)``` or ```% (e,)``` to make sure that the exception class isn't also a tuple (leading to an error where not all arguments are converted).  This same issue happens a few times below as well.